### ISSUE TITLE
Improving "the quite sad source code"

### ISSUE THREAD

**Issue by thomasdziedzic** [lobsters/lobsters#705]
The first line in the readme is this. It might put people off of using lobsters and contributing back!

One thing that has popped up over my interactions was the lack of testing that the code base has.

One potential solution might be to enable code climate to help with improving the quality of the code.

I checked codeclimate and it seems to be free for opensource projects, so that might be an option:
https://codeclimate.com/oss/

**Comment by pushcx** [comment]
I took the link as self-deprecating and tongue-in-cheek rather than harshly critical. (The original criticism, on the other hand...)

I've been slowly filling in missing tests (like 6c4bc0b42, which I mislaid and just recovered from my reflog), and I'm happy to have help.

I just tried to add CodeClimate, and it won't list the repo [because it's in an org](https://docs.codeclimate.com/docs/my-repository-isnt-listed-when-i-try-to-add-it-to-code-climate#section-open-source-repositories). I followed the steps to add it to the Lobsters org and resync (while constantly re-logging-in because it kept logging me out) but it didn't appear.

I'll leave this open if you want to contact support, but go ahead and close it if not.

**Comment by thomasdziedzic** [comment]
> I'll leave this open if you want to contact support, but go ahead and close it if not.

I can contact support but ultimately it would be up to you to troubleshoot it with codeclimate support since I don't have access to the lobsters organization.

If you want, I can send an email to their support and cc you on it.

Just to be clear, did you follow https://docs.codeclimate.com/docs/approve-code-climate-as-a-third-party-application-in-github and which steps did you complete vs have problems on?

**Comment by pushcx** [comment]
Yep. I had trouble on most of the steps. Aside from the constant log-outs, I got the impression it was running background tasks so I was often looking at, say, an empty list of repos and then reloading to see a partial list of my repos, etc. I'll give it another try when I have some more spare time.

**Comment by pushcx** [comment]
I'm grooming the backlog and I'm going to close out this old issue. We never did sort out CodeClimate, and I've very often heard from developers that they [enjoyed](https://i.kym-cdn.com/photos/images/newsfeed/000/897/718/c5f.gif) the self-deprecation.
